### ISSUE TITLE
[Spot] Inconsistency between sky stop and sky autostop

### ISSUE THREAD

**Issue by infwinston** [skypilot-org/skypilot#1472]
Found this during https://github.com/skypilot-org/skypilot/pull/1470, we do not support `sky stop` a spot VM but allow `sky autostop` it, which should be banned as well because its disk is not persistent on AWS?

```
(sky-tmp) weichiang@mbp sky % sky stop sky-2edf-weichiang
Stopping 1 cluster: sky-2edf-weichiang. Proceed? [Y/n]:
Stopping cluster 'sky-2edf-weichiang'... skipped.
  Stopping spot instances is not supported as the attached disks will be lost.
  To terminate the cluster instead, run: sky down sky-2edf-weichiang
Stopping 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% -:--:--


(sky-tmp) weichiang@mbp sky % sky autostop sky-2edf-weichiang
Scheduling autostop on 1 cluster: sky-2edf-weichiang. Proceed? [Y/n]:
Scheduling autostop on cluster 'sky-2edf-weichiang'...done
  The cluster will be autostopped after 5 minutes of idleness.
  To cancel the autostop, run: sky autostop sky-2edf-weichiang --cancel
Scheduling autostop on 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
```

**Comment by concretevitamin** [comment]
Good catch - we should ban it.

**Comment by mraheja** [comment]
Closing since spot autostop is no longer a thing.

**Comment by concretevitamin** [comment]
> Closing since spot autostop is no longer a thing.

Could you elaborate? I tried the snippet at the top and autostop for an unmanaged spot cluster still went through.

**Comment by concretevitamin** [comment]
Reopening, this is still an issue, and we should fix this before 0.3.0.

```
NAME          LAUNCHED   RESOURCES                                 STATUS  AUTOSTOP  COMMAND
dbg           < 1 sec    1x GCP(n2-standard-8[Spot])               UP      -         sky launch --cloud gcp -c dbg...

» sky autostop dbg
Scheduling autostop on 1 cluster: dbg. Proceed? [Y/n]:
Scheduling autostop on cluster 'dbg'...done
  The cluster will be autostopped after 5 minutes of idleness.
  To cancel the autostop, run: sky autostop dbg --cancel
Scheduling autostop on 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00

» sky stop dbg
Stopping 1 cluster: dbg. Proceed? [Y/n]:
Stopping cluster 'dbg'... skipped.
  Stopping spot instances is not supported as the attached disks will be lost.
  To terminate the cluster instead, run: sky down dbg
Stopping 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% -:--:--
```

**Comment by concretevitamin** [comment]
@mraheja @iojw @sumanthgenz @ewzeng have time to take this? Marking as P0 as we should fix it in a week or so to prep for 0.3 cut.

**Comment by ewzeng** [comment]
Hmm let me take a look
